js/ui/grid_core/ui.grid_core.editor_factory.js

```diff
@@ -2,497 +2,221 @@
+                that._updateFocusCore();
             }
+            that._isFocusOverlay = false;
+        });
+    },
 
-            that._createComponent($editorElement, options.editorName, options.editorOptions);
+    _updateFocusOverlaySize: function($element, position) {
+        var location = positionUtils.calculate($element, extend({ collision: "fit" }, position));
 
-            if(options.editorName === "dxTextBox") {
-                $editorElement.dxTextBox("instance").registerKeyHandler("enter", noop);
-            }
+        if(location.h.oversize > 0) {
+            $element.outerWidth($element.outerWidth() - location.h.oversize);
         }
-    };
-    return {
-        _getFocusedElement: function($dataGridElement) {
-            //T181706
-            return $dataGridElement.find(FOCUSED_ELEMENT_SELECTOR);
-        },
-
-        _getFocusCellSelector: function() {
-            return ".dx-row > td";
-        },
-
-        _updateFocusCore: function() {
-            var $focus = this._$focusedElement,
-                $dataGridElement = this.component && this.component.$element(),
-                $focusCell,
-                hideBorders;
-
-            if($dataGridElement) {
-                //this selector is specific to IE
-                $focus = this._getFocusedElement($dataGridElement);
-
-                if($focus.length) {
-                    if(!$focus.hasClass(CELL_FOCUS_DISABLED_CLASS)) {
-                        $focusCell = $focus.closest(this._getFocusCellSelector() + ", ." + CELL_FOCUS_DISABLED_CLASS);
-                        hideBorders = $focusCell.get(0) !== $focus.get(0) && $focusCell.hasClass(EDITOR_INLINE_BLOCK);
-                        $focus = $focusCell;
-                    }
-
-                    if($focus.length && !$focus.hasClass(CELL_FOCUS_DISABLED_CLASS)) {
-                        this.focus($focus, hideBorders);
-                        return;
-                    }
-                }
-            }
 
-            this.loseFocus();
-        },
-
-        _updateFocus: function(e) {
-            var that = this,
-                isFocusOverlay = e && e.jQueryEvent && $(e.jQueryEvent.target).hasClass(that.addWidgetPrefix(FOCUS_OVERLAY_CLASS));
+        if(location.v.oversize > 0) {
+            $element.outerHeight($element.outerHeight() - location.v.oversize);
+        }
+    },
 
-            that._isFocusOverlay = that._isFocusOverlay || isFocusOverlay;
+    callbackNames: function() {
+        return ["focused"];
+    },
 
-            clearTimeout(that._updateFocusTimeoutID);
+    focus: function($element, hideBorder) {
+        var that = this;
 
-            that._updateFocusTimeoutID = setTimeout(function() {
-                delete that._updateFocusTimeoutID;
-                if(!that._isFocusOverlay) {
-                    that._updateFocusCore();
-                }
-                that._isFocusOverlay = false;
-            });
-        },
+        if($element === undefined) {
```

Comment:
This `if-else` is a bit strange. 
What are other cases (neither `undefined` nor `~false`)?